src/olympia/translations/query.py

```diff
@@ -90,10 +90,16 @@ def clone(self):
         c.translation_aliases = self.translation_aliases
         return c
 
-    def get_compiler(self, using=None, connection=None):
+    def get_compiler(self, using=None, connection=None, **kwargs):
         # Call super to figure out using and connection.
-        c = super().get_compiler(using, connection)
-        return SQLCompiler(self, c.connection, c.using)
+        compiler = super().get_compiler(using, connection, **kwargs)
+        # slightly weird kwargs because django4.0 adds an extra elide_empty kwarg
+        return SQLCompiler(
+            self,
+            compiler.connection,
+            compiler.using,
+            **{kw: getattr(compiler, kw, None) for kw in kwargs if kw == 'elide_emtpy'},
+        )
```

Comment:
Instead of all of this, I suggest:
```diff
diff --git a/src/olympia/translations/query.py b/src/olympia/translations/query.py
index 5e9c48c072..a029e803aa 100644
--- a/src/olympia/translations/query.py
+++ b/src/olympia/translations/query.py
@@ -2,7 +2,7 @@ import itertools
 from collections import OrderedDict
 
 from django.conf import settings
-from django.db import models
+from django.db import connections, models
 from django.db.models.sql import compiler
 from django.db.models.sql.constants import LOUTER
 from django.db.models.sql.datastructures import Join
@@ -90,10 +90,16 @@ class TranslationQuery(models.sql.query.Query):
         c.translation_aliases = self.translation_aliases
         return c
 
-    def get_compiler(self, using=None, connection=None):
-        # Call super to figure out using and connection.
-        c = super().get_compiler(using, connection)
-        return SQLCompiler(self, c.connection, c.using)
+    def get_compiler(self, using=None, connection=None, **kwargs):
+        # Unfortunately even though there is a mechanism to override the
+        # compiler class through `self.compiler`, that needs to point to a
+        # member of the django.db.models.sql.compiler module (unless you're
+        # also customizing the database backend). So instead we do that
+        # here, returning our own custom SQLCompiler class. The parent
+        # implementation can set connection though, so we do that as well.
+        if using:
+            connection = connections[using]
+        return SQLCompiler(self, connection, using, **kwargs)
 
 
 class SQLCompiler(compiler.SQLCompiler):
```

Reply:
I looked into overriding the compiler class too - it's not that onerous as we already have a custom database backend for our `PositiveAutoField` field - the annoying part was that it wanted extra classes for Insert (i.e. `SQLInsertCompiler`), Delete and Update too, and I felt like the alternative ended up way more complicated than our current code.

Reply:
Ah, yeah, forgot we already had a custom backend. My comment should be reworded then to reflect what you just said, but I think the approach is still a bit better than trying to keep up with which parameters are present depending on the django version